cms/magento/magento2.php

```diff
@@ -0,0 +1,373 @@
+<?php
+
+/**
+ * Magento 2 Deployer recipe.
+ *
+ * @file
+ */
+
+declare(strict_types=1);
+
+namespace Deployer;
+
+use Deployer\Exception\RunException;
+
+require_once __DIR__ . '/../../src/functions.php';
+import('recipe/common.php');
+
+set('app_type', 'magento');
+set('mage', 'bin/magento');
+fill('shared_dirs', [
+    'var/composer_home',
+    'var/log',
+    'var/export',
+    'var/report',
+    'var/import',
+    'var/import_history',
+    'var/session',
+    'var/importexport',
+    'var/backups',
+    'var/tmp',
+    'pub/sitemap',
+    'pub/media',
+]);
+fill('shared_files', [
+    'app/etc/env.php',
+    'var/.maintenance.ip',
+]);
+fill('writable_dirs', [
+    'var',
+    'pub/static',
+    'pub/media',
+    'generated',
+    'pub/page-cache',
+]);
+fill('clear_paths', [
+    'generated/*',
+    'pub/static/_cache/*',
+    'var/generation/*',
+    'var/cache/*',
+    'var/page_cache/*',
+    'var/view_preprocessed/*',
+]);
+fill('app_directory_name', 'docroot');
+set('static_content_locales', 'en_US');
+set('http_user', 'www-data');
+//set('writable_use_sudo',true);
+fill('writable_recursive', 'true');
+
+import('vendor/unleashedtech/deployer-recipes/config.php');
+
+/**
+ * Initialize writable_dirs variable.
```

Comment:
Given recent permissions discussions, perhaps we should avoid having Deployer set directory permissions? It could still be _supported_, but not used by default. If we confirm standard practice is that only _writable_ directories are those that are _shared_ among the deployments, permissions can be set _once_ on those shared directories. Thoughts?

Reply:
I agree the writable_directories and shared_directories implementation needs work. It's working for our magento2 current site only because of this setup AND some significant backend DevOps work that is not seen here.  So I'm not happy about that aspect. My preference is to have everything permissions related in the same configuration file and I'm still hopeful we can achieve that with the talked about perms of ( 
`chown -R www-data:www-data /sites/default/files
find /sites/default/files -type d -print0 | xargs -0 chmod u+rwxg+rwso+rx-w
find /sites/default/files -type f -print0 | xargs -0 chmod 644`